frontend/src/pages/documents/DocumentsList.vue

```diff
@@ -0,0 +1,191 @@
+            </document-item>
+          </ul>
+          <h2 v-else>You have no documents yet</h2>
+        </base-card>
+      </section>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import DocumentItem from '../../components/documents/document-item.vue';
+import BaseCard from '../../components/cards/card.vue';
+import BaseButton from '../../components/buttons/link-button.vue';
+import BaseDialog from '../../components/dialogs/dialog.vue';
+
+export default {
+  components: {
+    DocumentItem,
+    BaseCard,
+    BaseButton,
+    BaseDialog,
+  },
+  data(): { [prop: string]: any } {
+    return {
+      createDialogOpened: false,
+      title: '',
+      deleteDialogOpened: false,
+      documentID: '',
+      contributorDialogOpened: false,
+      userName: '',
+      emptyTitle: false,
+      emptyUserName: false,
+    };
+  },
+  computed: {
+    documents(): { [prop: string]: string | string[] } {
+      return this.$store.getters['documents/documents'];
+    },
+    hasDocuments(): boolean {
+      return this.$store.getters['documents/hasDocuments'];
+    },
+  },
+  methods: {
+    toggleCreateDialog(): void {
+      this.createDialogOpened = !this.createDialogOpened;
+    },
+    toggleDeleteDialog(id: string = ''): void {
+      if (id) this.documentID = id;
+      this.deleteDialogOpened = !this.deleteDialogOpened;
+    },
+    toggleContributorDialog(id: string = ''): void {
+      if (id) this.documentID = id;
+      this.contributorDialogOpened = !this.contributorDialogOpened;
+    },
+    createdDocument(): void | undefined {
+      if (this.title === '') {
+        this.emptyTitle = true;
+        return;
+      }
+      this.emptyTitle = false;
+      this.toggleCreateDialog();
+      const id = this.createID();
+      const document = {
+        id: id,
+        title: this.title,
+        contributors: [this.$store.getters['auth/userName']],
+        createdDate: this.dateAndTime(),
+        modifiedDate: this.dateAndTime(),
+      };
+      this.$store.dispatch('documents/createDocument', document);
+      this.title = '';
+    },
+    deleteDocument(): void {
+      this.toggleDeleteDialog();
+      const index = this.$store.getters['documents/documents']
+        .findIndex(value => value.id === this.documentID);
+      this.$store.dispatch('documents/deleteDocument', index);
+    },
+    addContributor(): void | undefined {
+      if (this.userName === '') {
+        this.emptyUserName = true;
+        return;
+      }
+      this.emptyUserName = false;
+      this.toggleContributorDialog();
+      const index = this.$store.getters['documents/documents']
+        .findIndex(value => value.id === this.documentID);
+      this.$store.dispatch('documents/addContributor', [index, this.userName]);
+      this.changeModifiedDate(index);
+      this.userName = '';
+    },
+    dateAndTime(): string {
+      const date = new Date();
```

Comment:
Move to shared or include library for working with times (please not moment - https://github.com/you-dont-need/You-Dont-Need-Momentjs)